tests/test_hook_manager.py

```diff
@@ -291,6 +291,23 @@ def inst_hook(zelos, address, size):
         z.start()
         self.assertEqual(z.regs.getIP(), 0x8048B72)
 
+    def test_func_hook(self):
```

Comment:
Include another test that hooks a function in the middle of a run, after the target module has been loaded into memory. For example, hook `malloc` after it has already been called once and see if the remaining call to malloc is hooked successfully.

Reply:
Addressed in an update to the test. Added ability to set a condition to delete the hook as well.